test/test_credential.js

```diff
@@ -0,0 +1,245 @@
+        { from: accounts[1], value: oneEth.dividedBy(100) },
+      ),
+      'Student number cannot be empty',
+    );
+    await truffleAssert.reverts(
+      credentialInstance.addCredential(
+        'Lyn Tan',
+        'A0123456L',
+        '',
+        'Bachelor of Computing',
+        'Dr Li Xiaofan',
+        0, // Institution ID
+        toUnixTime(2023, 3, 21), // Issuance date
+        0, // Expiry date
+        accounts[7], // Student A,
+        { from: accounts[1], value: oneEth.dividedBy(100) },
+      ),
+      'Course name cannot be empty',
+    );
+    await truffleAssert.reverts(
+      credentialInstance.addCredential(
+        'Lyn Tan',
+        'A0123456L',
+        'Information Systems',
+        '',
+        'Dr Li Xiaofan',
+        0, // Institution ID
+        toUnixTime(2023, 3, 21), // Issuance date
+        0, // Expiry date
+        accounts[7], // Student A,
+        { from: accounts[1], value: oneEth.dividedBy(100) },
+      ),
+      'Degree level cannot be empty',
+    );
+    await truffleAssert.reverts(
+      credentialInstance.addCredential(
+        'Lyn Tan',
+        'A0123456L',
+        'Information Systems',
+        'Bachelor of Computing',
+        '',
+        0, // Institution ID
+        toUnixTime(2023, 3, 21), // Issuance date
+        0, // Expiry date
+        accounts[7], // Student A,
+        { from: accounts[1], value: oneEth.dividedBy(100) },
+      ),
+      'Endorser name cannot be empty',
+    );
+    await truffleAssert.reverts(
+      credentialInstance.addCredential(
+        'Lyn Tan',
+        'A0123456L',
+        'Information Systems',
+        'Bachelor of Computing',
+        'Dr Li Xiaofan',
+        0, // Institution ID
+        0, // Issuance date
+        0, // Expiry date
+        accounts[7], // Student A,
+        { from: accounts[1], value: oneEth.dividedBy(100) },
+      ),
+      'Issuance date cannot be empty',
+    );
+    await truffleAssert.reverts(
+      credentialInstance.addCredential(
+        'Lyn Tan',
+        'A0123456L',
+        'Information Systems',
+        'Bachelor of Computing',
+        'Dr Li Xiaofan',
+        0, // Institution ID
+        toUnixTime(2023, 8, 1), // Issuance date
+        0, // Expiry date
+        accounts[7], // Student A,
+        { from: accounts[1], value: oneEth.dividedBy(100) },
+      ),
+      'Issuance date cannot be a future date. Please enter an issuance date that is today or in the past.',
+    );
+    await truffleAssert.reverts(
+      credentialInstance.addCredential(
+        'Lyn Tan',
+        'A0123456L',
+        'Information Systems',
+        'Bachelor of Computing',
+        'Dr Li Xiaofan',
+        0, // Institution ID
+        toUnixTime(2023, 3, 21), // Issuance date
+        0, // Expiry date
+        zeroAddress, // Student A,
+        { from: accounts[1], value: oneEth.dividedBy(100) },
+      ),
+      'Student address cannot be empty',
+    );
+  });
+
+  it('Delete Credential', async () => {
```

Comment:
perhaps instead of just emitting an event, we can test if the credential after deletion has the credentialState DELETED. same for revoked